In [1]:
import pandas as pd


In [2]:
arquivo = r"../data/raw/Preços semestrais - AUTOMOTIVOS_2026.01.csv"

df = pd.read_csv(
    arquivo,
    sep=";",
    encoding="utf-8"
)

df.head()

,Regiao - Sigla,Estado - Sigla,Municipio,Revenda,CNPJ da Revenda,Nome da Rua,Numero Rua,Complemento,Bairro,Cep,Produto,Data da Coleta,Valor de Venda,Valor de Compra,Unidade de Medida,Bandeira
0,N,AC,CRUZEIRO DO SUL,AMAZONIA COMERCIO DE DERIVADOS DE PETROLEO LTDA,01.492.748/0003-83,AVENIDA COPACABANA,440,NaN,COPACABANA,69980-000,GASOLINA,02/01/2026,"7,97",NaN,R$ / litro,IPIRANGA
1,N,AC,CRUZEIRO DO SUL,AMAZONIA COMERCIO DE DERIVADOS DE PETROLEO LTDA,01.492.748/0003-83,AVENIDA COPACABANA,440,NaN,COPACABANA,69980-000,DIESEL,02/01/2026,"8,15",NaN,R$ / litro,IPIRANGA
2,N,AC,CRUZEIRO DO SUL,AMAZONIA COMERCIO DE DERIVADOS DE PETROLEO LTDA,01.492.748/0003-83,AVENIDA COPACABANA,440,NaN,COPACABANA,69980-000,DIESEL S10,02/01/2026,"8,17",NaN,R$ / litro,IPIRANGA
3,N,AC,RIO BRANCO,AUTO POSTO AMAPA LTDA,00.529.581/0001-53,VIA CHICO MENDES,3570,NaN,AREAL,69906-119,GASOLINA,02/01/2026,"7,29",NaN,R$ / litro,VIBRA
4,N,AC,RIO BRANCO,AUTO POSTO AMAPA LTDA,00.529.581/0001-53,VIA CHICO MENDES,3570,NaN,AREAL,69906-119,GASOLINA ADITIVADA,02/01/2026,"7,39",NaN,R$ / litro,VIBRA


In [3]:
print("Linhas:", df.shape[0])
print("Colunas:", df.shape[1])


Linhas: 422418
Colunas: 16


In [4]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 422418 entries, 0 to 422417
Data columns (total 16 columns):
 #   Column             Non-Null Count   Dtype  
---  ------             --------------   -----  
 0   Regiao - Sigla     422418 non-null  str    
 1   Estado - Sigla     422418 non-null  str    
 2   Municipio          422418 non-null  str    
 3   Revenda            422418 non-null  str    
 4   CNPJ da Revenda    422418 non-null  str    
 5   Nome da Rua        422418 non-null  str    
 6   Numero Rua         422412 non-null  str    
 7   Complemento        96159 non-null   str    
 8   Bairro             421690 non-null  str    
 9   Cep                422418 non-null  str    
 10  Produto            422418 non-null  str    
 11  Data da Coleta     422418 non-null  str    
 12  Valor de Venda     422418 non-null  str    
 13  Valor de Compra    0 non-null       float64
 14  Unidade de Medida  422418 non-null  str    
 15  Bandeira           422418 non-null  str    
dtypes: float64(1)

In [6]:
df.columns.tolist()

['Regiao - Sigla',
 'Estado - Sigla',
 'Municipio',
 'Revenda',
 'CNPJ da Revenda',
 'Nome da Rua',
 'Numero Rua',
 'Complemento',
 'Bairro',
 'Cep',
 'Produto',
 'Data da Coleta',
 'Valor de Venda',
 'Valor de Compra',
 'Unidade de Medida',
 'Bandeira']

In [7]:
# Quantidade e percentual de valores ausentes por coluna

valores_ausentes = pd.DataFrame({
    "quantidade_ausente": df.isna().sum(),
    "percentual_ausente": (df.isna().mean() * 100).round(2)
})

valores_ausentes.sort_values(
    by="quantidade_ausente",
    ascending=False
)

,quantidade_ausente,percentual_ausente
Valor de Compra,422418,100.00
Complemento,326259,77.24
Bairro,728,0.17
Numero Rua,6,0.00
Revenda,0,0.00
Municipio,0,0.00
Regiao - Sigla,0,0.00
Estado - Sigla,0,0.00
CNPJ da Revenda,0,0.00
Nome da Rua,0,0.00


In [8]:
# Quantidade de linhas completamente duplicadas

quantidade_duplicadas = df.duplicated().sum()

print("Linhas duplicadas:", quantidade_duplicadas)
print(
    "Percentual de duplicadas:",
    round(quantidade_duplicadas / len(df) * 100, 2),
    "%"
)

Linhas duplicadas: 6
Percentual de duplicadas: 0.0 %


In [9]:
# Valores distintos das principais colunas categóricas

print("Regiões:", sorted(df["Regiao - Sigla"].dropna().unique()))
print("Estados:", df["Estado - Sigla"].nunique())
print("Municípios:", df["Municipio"].nunique())
print("Revendas:", df["CNPJ da Revenda"].nunique())
print("Produtos:", sorted(df["Produto"].dropna().unique()))
print("Bandeiras:", df["Bandeira"].nunique())

Regiões: ['CO', 'N', 'NE', 'S', 'SE']
Estados: 27
Municípios: 416
Revendas: 7963
Produtos: ['DIESEL', 'DIESEL S10', 'ETANOL', 'GASOLINA', 'GASOLINA ADITIVADA', 'GNV']
Bandeiras: 47


In [10]:
# Exemplos dos campos que serão convertidos posteriormente

df[[
    "Data da Coleta",
    "Valor de Venda",
    "Valor de Compra",
    "Unidade de Medida"
]].head(10)

,Data da Coleta,Valor de Venda,Valor de Compra,Unidade de Medida
0,02/01/2026,"7,97",NaN,R$ / litro
1,02/01/2026,"8,15",NaN,R$ / litro
2,02/01/2026,"8,17",NaN,R$ / litro
3,02/01/2026,"7,29",NaN,R$ / litro
4,02/01/2026,"7,39",NaN,R$ / litro
5,02/01/2026,"7,35",NaN,R$ / litro
6,02/01/2026,"5,24",NaN,R$ / litro
7,02/01/2026,"7,29",NaN,R$ / litro
8,02/01/2026,"7,39",NaN,R$ / litro
9,02/01/2026,"7,35",NaN,R$ / litro


## Conclusões do diagnóstico inicial

A base pública da ANP referente aos preços de combustíveis automotivos possui 422.418 registros e 16 colunas.

### Cobertura da base

- 5 regiões
- 27 estados
- 416 municípios
- 7.963 revendas identificadas por CNPJ
- 47 bandeiras
- 6 produtos

Os produtos encontrados foram:

- Diesel
- Diesel S10
- Etanol
- Gasolina
- Gasolina Aditivada
- GNV

### Qualidade dos dados

Foram identificadas 6 linhas completamente duplicadas, correspondendo a aproximadamente 0,0014% da base.

A coluna `Valor de Compra` possui 100% dos registros ausentes e não poderá ser utilizada nas análises.

A coluna `Complemento` possui 77,24% de valores ausentes. Essa ausência é esperada para dados de endereço e não compromete as análises de preço.

Também foram encontradas pequenas quantidades de valores ausentes nas colunas `Bairro`, `Numero Rua` e `Revenda`.

### Tipos de dados

A coluna `Data da Coleta` foi carregada como texto e deverá ser convertida para data.

A coluna `Valor de Venda` também foi carregada como texto devido à utilização da vírgula como separador decimal e deverá ser convertida para um tipo numérico.

### Próximas ações

- Padronizar os nomes das colunas.
- Converter a data da coleta.
- Converter o valor de venda para número decimal.
- Remover a coluna de valor de compra.
- Avaliar as unidades de medida por produto.
- Validar e remover duplicidades completas.
- Exportar uma base tratada sem modificar o arquivo original.